keg_auth/testing.py

```diff
@@ -714,9 +714,9 @@ def test_next_parameter_not_open_redirect(self):
 
         assert resp.status_code == 302, resp.html
         # verify the 'next' parameter was ignored
-        assert resp.headers['Location'] == flask.url_for(
+        assert resp.headers['Location'] == urlparse(flask.url_for(
```

Comment:
These tests are an anti-pattern.  We should be testing against a concrete expected value, not re-implementing the logic we use to calculate the URL in the main code.  E.g.:

```python
assert resp.headers['Location'] == '/login'
```

@guruofgentoo unless you object today, @calebsyring can make these adjustments while he's at it.

Reply:
Some injection either on the front side (mocking) or the tail end (as it is here), because this set of tests is meant to be run in any of our projects, which may have differing paths for each auth endpoint. I think putting that injection on the front side is cleaner, since it simplifies the assert.

Reply:
But I think the point of these tests is to exercise the auth setup, so I don't think we'd want to mock the requests, would we?  Or, what would we be mocking?

Maybe what we should be doing here is having the app's override this base class and set hard-coded variables on the subclass to represent expected URLs.  Then, in this class, we'd do something like `assert resp.headers['Location'] == self.LOGIN_URL`.

```python
# in the app's test_auth.py
class TestAuthIntegration(AuthTests):
    user_ent = ents.User
    protected_url = '/users'
    protected_url_permissions = 'auth-manage'
   LOGIN_URL = '/login'
   FOO_URL = '/bar'
```

Thoughts?

Reply:
It'd essentially be a "config"-style mock on the auth manager to make the endpoint correspond to a known value.

Reply:
@guruofgentoo, how would that work? Since the tests are meant to be run in any of our projects, how could we guarantee the endpoint would exist without subclass attributes?

Reply:
On further thought, the above is correct - class attributes is the way to go here.